### PR TITLE
Assessment plan

### PR DESCRIPTION
**Pull request by ChristianMurphy**

### CODE REVIEW

File: grails-app/controllers/imod/AssessmentTechniqueController.groovy [hunk 1]
```diff
@@ -96,10 +97,12 @@ class AssessmentTechniqueController {
 
 		// This checks when a technique is assigned to a learning objective
 
+
 		if(params.assigncheck == true) {
-							// get current user object
-		def currentLearningObjective = LearningObjective.findById(learningObjectiveID)
+		//println("goes into asigncheck if:: "+params.assigncheck )
```
Comment (ChristianMurphy) [inline grails-app/controllers/imod/AssessmentTechniqueController.groovy]: The code inside the `if` block.
between the `{}` should be tabbed one additional level.

File: grails-app/controllers/imod/AssessmentTechniqueController.groovy [hunk 2]
```diff
@@ -107,6 +110,7 @@ class AssessmentTechniqueController {
 		currentLearningObjective.save()
 		}
 
+
 		// This checks when a technique is favoritized  to by a user
 		if(params.favcheck == true) {
 		// get current user object
```
Comment (ChristianMurphy) [inline grails-app/controllers/imod/AssessmentTechniqueController.groovy]: The code inside the `if` block.
between the `{}` should be tabbed one additional level.

File: grails-app/controllers/imod/AssessmentTechniqueController.groovy [hunk 3]
```diff
@@ -167,15 +171,16 @@ class AssessmentTechniqueController {
 		// persist new technique to database
 		newTechnique.save()
 
-// This checks when a technique is assigned to a learning objective
 
 		if(params.assigncheck == true) {
 		// get current user object
```
Comment (ChristianMurphy) [inline grails-app/controllers/imod/AssessmentTechniqueController.groovy]: The code inside the `if` block.
between the `{}` should be tabbed one additional level.

File: grails-app/views/assessment/index.gsp [hunk 4]
```diff
@@ -215,6 +215,10 @@
 						<%--To render the add new Technique dialog box--%>
 
 						<g:form controller="assessmentTechnique" method="post" id="${currentImod.id}" params="[learningObjectiveID: currentLearningObjective.id]">
+						<g:javascript>
+							var currentLO = ${currentLearningObjective.id};
```
Comment (ChristianMurphy) [inline grails-app/views/assessment/index.gsp]: What is this for?

File: grails-app/views/assessment/index.gsp [hunk 5]
```diff
@@ -385,37 +389,20 @@
...
-					<div id="assessment-plan" >
-						<g:form controller="assessmentTechnique" method="post" id="${currentImod.id}" params="[learningObjectiveID: currentLearningObjective.id]">
-							<label>
-								<h2>
-									Learning Objective
-								</h2>
-							</label>
-							<br/>
+<%--Dialog box for Assessement Plans--%>
+<div id="assessment-plan" >
```
Comment (ChristianMurphy) [inline grails-app/views/assessment/index.gsp]: this needs to be tabbed
this `div` should be on the same tab level as the `div` on line `390`

File: grails-app/views/assessment/index.gsp [hunk 6]
```diff
@@ -385,37 +389,20 @@
...
+<g:form controller="assessmentTechnique" method="post" id="${currentImod.id}" params="[learningObjectiveID: currentLearningObjective.id]">
```
Comment (ChristianMurphy) [inline grails-app/views/assessment/index.gsp]: Each tag should have its own line
``` html
bad
<div><div></div></div>

good
<div>
    <div></div>
</div>
```

And each time there is a tag inside a tag it should be tabbed
``` html
bad
<div>
<div></div>
</div>

good
<div>
    <div></div>
</div>
```

File: web-app/css/source/assessment.css [hunk 7]
```diff
@@ -13,14 +13,13 @@
     font-size: 1em;}
 .form-labels { display: table-cell; width: 150px;    font-family: helvetica;
     font-size: 1em;}
-.form-inputs { display: table-cell; width: 200px;     font-family: helvetica;
+.form-inputs { display: table-cell; width: 400px;     font-family: helvetica;
```
Comment (ChristianMurphy) [inline web-app/css/source/assessment.css]: rules should be on a new line after selector, and each rule should have its own line, rules should be tabbed within selector context.
``` css
bad
.example { display: none; color: blue; }

good
.example {
   display: none;
   color: blue;
}
```

File: web-app/js/source/assessment.js [hunk 8]
```diff
@@ -169,21 +170,42 @@ function filterAssessmentTechniques() {
 }
 
 function checkForAssign(data) {
-	// Your code
+	var assignedLO='';
 	for (var index = 0; index < data.idealAssessmentTechniqueMatch.length; index++) {
 		var currentTechnique = data.idealAssessmentTechniqueMatch[index];
+	//alert("for loop for ideal tech");
```
Comment (ChristianMurphy) [inline web-app/js/source/assessment.js]: remove the debug `alert`s

File: web-app/js/source/assessment.js [hunk 9]
```diff
@@ -319,47 +386,106 @@ function assessmentPlanData(data) {
...
 	$('#assessmentData').html(allAssessmentData);
 
 }
 
+
+
+
+function checkIndex(){
+
+}
```
Comment (ChristianMurphy) [inline web-app/js/source/assessment.js]: empty `function` block?

**Comment by ChristianMurphy** [comment]
The `assessment.js` may need a bit of cleanup before it can be merged.
`JSHint` and `JSCS` will make this pretty easy.
just run
``` sh
npm install -g jscs jshint
```
and then run the checkers against the `assessment.js`
```
cd /path/to/project
jshint web-app/js/source/assessment.js
jscs web-app/js/source/assessment.js
```
and it will show the code changes that should be made.

**Comment by ChristianMurphy** [comment]
@hgafoor28 re-opened PR against `development` branch.
Resolve the merge conflict and a little bit of code cleanup and this should be good to go! :smile:

**Comment by hgafoor28** [comment]
Hi Christian,

I'll get to it when I am done with work later today. Thanks for the update.

--Husna

On Thu, May 14, 2015 at 11:37 AM, Christian Murphy <notifications@github.com

**Comment by ChristianMurphy** [comment]
@hgafoor28 Any updates on this?

**Comment by ChristianMurphy** [comment]
This PR is several sprints behind, and has had no update since May.
Closing it out.

**Pull request closed without merging**